src/anchor_in_depth/Events.md

```diff
@@ -0,0 +1,92 @@
+# Events in Anchor Solana
+
+Events in Anchor Solana provide a powerful mechanism for notifying and communicating between different components of a Solana dApp. They allow for the emission and tracking of significant occurrences within the program's execution. This documentation will cover the concept of events in Anchor Solana, their structure, and how to use them effectively in your smart contract development.
+
+## Table of Contents
+- [Introduction to Events](#introduction-to-events)
+- [Defining Events](#defining-events)
+- [Emitting Events](#emitting-events)
+- [Subscribing to Events](#subscribing-to-events)
+
+## Introduction to Events
+An event is a structured piece of data that holds information about a specific occurrence in a smart contract program. It acts as a message to notify external parties about significant changes or transactions within the contract's state. Events can be used to provide transparency, traceability, and synchronization in decentralized applications. 
+
+Events in Anchor are basically base64 encoded structs that are program logged. For ex, when you may write msg!() something to debug your Solana program, you are logging data. Events are essentially the same thing but they are not human readable as they are base64 encoded. The reason for this is that it is quite expensive to log string data:
+
+```rust
+#[proc_macro]
+pub fn emit(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let data: proc_macro2::TokenStream = input.into();
+    proc_macro::TokenStream::from(quote! {
+        {
+            anchor_lang::solana_program::log::sol_log_data(&[&anchor_lang::Event::data(&#data)]);
+        }
+    })
+}
+```
+
+## Defining Events
+In Anchor Solana, events are defined using the `#[event]` attribute macro. This macro allows you to specify the fields that an event should contain. Events can include various data types, making them versatile for different use cases.
+
+```rust
+use anchor_lang::prelude::*;
+
+#[event]
+pub struct TransferEvent {
+    from: Pubkey,
+    to: Pubkey,
+    amount: u64,
+}
+```
+
+In this example, we define an event named `TransferEvent` with three fields: `from` (sender's address), `to` (receiver's address), and `amount` (the transferred amount).
+
+## Emitting Events
+To emit an event within your Anchor Solana program, you can use the `emit!` macro. This allows you to create an instance of the event structure and broadcast it to the network.
```

Comment:
> This allows you to create an instance of the event structure and broadcast it to the network.

Using `emit!` does not allow creating an instance of the event structure as they are separate actions.